backends/platform/sdl/posix/posix.cpp

```diff
@@ -82,11 +81,54 @@ bool OSystem_POSIX::hasFeature(Feature f) {
 Common::String OSystem_POSIX::getDefaultConfigFileName() {
 	Common::String configFile;
 
-	// On POSIX type systems, by default we store the config file inside
-	// to the HOME directory of the user.
-	const char *home = getenv("HOME");
-	if (home != NULL && (strlen(home) + 1 + _baseConfigName.size()) < MAXPATHLEN) {
-		configFile = Common::String::format("%s/%s", home, _baseConfigName.c_str());
+	Common::String prefix;
+#ifdef MACOSX
+	prefix = getenv("HOME");
+#elif !defined(SAMSUNGTV)
+	const char *envVar;
+	// Our old configuration file path for POSIX systems was ~/.scummvmrc.
```

Comment:
That is an option. However, I think it is annoying if you work with older versions of ScummVM for testing. You would end up having to maintain two configuration files. Thus, I am always using `~/.scummvmrc` if that exists.

If we go for auto-upgrading never the less, we will need to make sure we don't override the new one if it exists. Otherwise things can easily break when `~/.scummvmrc` is present (or gets created again by running an older version of ScummVM).